modules/vshn-lbaas-exoscale/hiera.tf

```diff
@@ -33,8 +46,10 @@ module "hiera" {
   lb_api_credentials = {
     cloudscale = null
     exoscale = {
-      key    = var.lb_exoscale_api_key
-      secret = var.lb_exoscale_api_secret
+      # Use `nonsensitive` so we get useful apply diffs for the hieradata (if
+      # we already have a copy available locally).
+      key    = nonsensitive(exoscale_iam_access_key.floaty.key)
+      secret = nonsensitive(exoscale_iam_access_key.floaty.secret)
```

Comment:
We could leave out the `nonsensitive()` here, which would result in `terraform apply` output which looks somewhat like:

```
  # module.cluster.module.lb.module.hiera[0].local_file.lb_hieradata must be replaced
-/+ resource "local_file" "lb_hieradata" {
      ~ content              = (sensitive) # forces replacement
      ~ id                   = "53055da8c1e5f86abcf295c045a40883413aabdc" -> (known after apply)
        # (3 unchanged attributes hidden)
    }
```

compared to (with `nonsensitive()`):

```
  # module.cluster.module.lb.module.hiera[0].local_file.lb_hieradata must be replaced
-/+ resource "local_file" "lb_hieradata" {
      ~ content              = <<-EOT # forces replacement
            # Managed by Terraform for Project Syn cluster c-autumn-sun-5704
            profile_openshift4_gateway::nodes:
              - lb-4e.c-autumn-sun-5704.appuio-beta.ch
              - lb-4a.c-autumn-sun-5704.appuio-beta.ch
            profile_openshift4_gateway::public_interface: eth0
            profile_openshift4_gateway::private_interfaces:
              - eth1
            profile_openshift4_gateway::floating_addresses:
              api: 194.182.191.63
              router: 194.182.165.105
            profile_openshift4_gateway::floating_address_provider: exoscale
            profile_openshift4_gateway::floating_address_settings:
              key: EXO.....
              secret: P6......
            profile_openshift4_gateway::backends:
              'api':
                - etcd-0.c-autumn-sun-5704.appuio-beta.ch
                - etcd-1.c-autumn-sun-5704.appuio-beta.ch
                - etcd-2.c-autumn-sun-5704.appuio-beta.ch
              'router':
                - 194.182.191.244
                - 194.182.164.138
                - 194.182.188.167
          - profile_openshift4_gateway::enable_proxy_protocol: true
            profile_icinga2::host::hiera_vars:
              team: tarazed
        EOT
      ~ id                   = "53055da8c1e5f86abcf295c045a40883413aabdc" -> (known after apply)

```

Unfortunately, Terraform doesn't support keeping a part of a string value sensitive (so we can't get readable output for unrelated changes with the sensitive values replaced with `*`), so if we leave the api key and secret marked sensitive, the whole hiera contents are marked sensitive.

Reply:
The key will be in git anyways so not much more exposure if it's in CI logs too i guess 🤷‍♀️